### ISSUE TITLE
Bug in Radia VCFs without matched RNA-seq

### ISSUE THREAD

**Issue by ckandoth** [covingto/pancanmafmerge#7]
A ton of radia calls have a 100% VAF, with ExAC tagging most of them as germline. These calls have SS=1 (i.e. germline) but FILTER=PASS. On closer look, it's limited to a handful of TN-pairs that did not have matched RNA-seq. Here are some examples...

```
vcfs/LUAD/abb86d09-d4ff-41f4-ba22-185dc48347c9_01_11/filtered.radia.tcga_filtered.reheadered.vcf
vcfs/LAML/61db06eb-299a-48b3-8c71-4bea78169b4c_03_11/filtered.radia.tcga_filtered.reheadered.vcf
vcfs/LAML/b4bc4666-13ba-4864-be9b-89ff8a0a1bd8_03_11/filtered.radia.tcga_filtered.reheadered.vcf
```

**Comment by covingto** [comment]
Can these be reprocessed with a filter for radia. Any way to identify the
VCFs affected? We are selecting only based on pass not SS.

**Comment by ckandoth** [comment]
I've only checked a handful, but they all seem to be Radia VCFs without the `RNA_PRIMARY` sample as a genotype column

**Comment by ckandoth** [comment]
A regex like this should do the trick...

```
perl -i -a -F'\t' -pe 'F[6]=Germline if(!m/^#/ and $F[7]=~m/SS=1/); $_=join("\t",@F)' */*/filtered.radia.tcga_filtered.reheadered.vcf
```

**Comment by covingto** [comment]
perl isn't really my language, but it looks like this just converts the
record to indicate filter is Germline with an in-place modification.

I'm finding the files with lines that have SS=1 and PASS in them.  Seems
that there are quite a few though.  I think we can still deliver on Monday
if we redo these but it might be close.  Need to to loop Amie and the UCSC
group in on this.

**Comment by ckandoth** [comment]
No worries. Python can do regexes too, but I'm not sure where to put this in the code. It will also add a `##FILTER` line in the VCF header...

```
import sys,re

for line in sys.stdin:

    # Add a new ##FILTER line in the VCF header
    if re.match( '#CHROM', line ):
        line = "##FILTER=<ID=germ,Description=\"Germline call\">\n" + line;

    # For all lines with variant calls, set FILTER=germ if INFO/SS=1 or INFO/MT=GERM
    if not re.match( '#', line ):
        cols = line.split('\t')
        if re.search( 'SS=1|MT=GERM', cols[7] ):
            cols[6] = 'germ'
        line = '\t'.join( cols )
    sys.stdout.write( line )
```

I'm actually considering removing these calls outright because they're not even reported in the other radia VCFs. Would love to hear thoughts from Amie, @kellrott and @singerma.

**Comment by covingto** [comment]
It's ok. I'm one step ahead of you. I made filter_radia.py which reads and
only prints if the line begins with #, or does not contain SS=1 and PASS.

It's in the filter block of merge.py. Not checked in yet. But if you want
to filter anything or otherwise manipulate the input VCF you should make a
separate application and call that from filter.

Thanks
Kyle

**Comment by ckandoth** [comment]
Nice. Thanks. That's what I prefer as well. Toss those calls. They're in the hundreds of thousands!

**Comment by covingto** [comment]
Yeah. This should prolly be a bug report to radia.

For filtering you can write the filter program in any language you want and
we call it from filter.

**Comment by covingto** [comment]
Yeah. This should prolly be a bug report to radia.

For filtering you ca write the filter program in any language you want and
we call it from filter.

Fixes pushed up and on tag v0.2.

Kyle
